### PR TITLE
Add multi-site support.

### PR DESCRIPTION
**Pull request by sjpadgett**
Fix ACL authorizations.
We need to discuss how acl.inc is used outside of normal login authentication. Strange.

### CODE REVIEW

**Comment by sjpadgett** [comment]
@MatthewVita  I just went ahead and created a branch for this update so it can be reviewed before a merge with master. I've tested against two multi-sites plus default.

**Comment by ghost** [comment]
Are there tasks I can take to work on?

Jerry, Matt, please let me know.

**Comment by sjpadgett** [comment]
@danehrlich1 You might want to clone this git and maybe do some testing to get a feel. Then maybe take on a controller/service which is a pretty stable non changing architecture in the project just now. The dispatcher is still evolving with the next step of probably a full login for the user meaning, some namespace and class moves to accommodate. 
Or write a small client test jig app to do authentication and test various routes. It is a pain using browser REST clients for the auth handshake.

File: apis/dispatch.php [hunk 1]
```diff
@@ -40,12 +31,34 @@
...
+    // Get a site id from initial log in authentication.
+    $data = (array)(json_decode(file_get_contents("php://input")));
+    $site = empty($data['site']) ? "default" : $data['site'];
+    $_GET['site'] = $site;
+} else {
+    if (strlen($_SERVER["HTTP_X_API_TOKEN"]) > 32) {
+        $token = str_split($_SERVER["HTTP_X_API_TOKEN"], 32);
+        $_SERVER["HTTP_X_API_TOKEN"] = $token[0]; // reset hash to further the adventure.
+        $_GET['site'] = hex2bin($token[1]); // site id
+    }
```
Comment (bradymiller) [inline apis/dispatch.php:46]: do you need an else here to set the site to default (in case the token is size 32)?

File: apis/dispatch.php [hunk 2]
```diff
@@ -65,6 +78,9 @@ function authorization_check($section, $value)
     }
 }
 
-// authentication_check($resource);
+
+authentication_check($resource);
 
 HttpRestRouteHandler::dispatch($routes, $resource, $_SERVER["REQUEST_METHOD"]);
+// Tear down session for security.
+$gbl->destroySession();
```
Comment (bradymiller) [inline apis/dispatch.php:86]: yes, destruction, destruction, destruction :)

**Comment by bradymiller** [comment]
Very nice. Looks like you've fixed both multisite and acl_check in one swoop. What's the best way to test this (ie. is there a tool folks use to test api environments)?

**Comment by sjpadgett** [comment]
Hi @bradymiller Thanks for the review. These days I really do need somebody to keep me honest:) That's why I let you do a once over before I do any merges...
The way I test REST is with a browser plugin. For Firefox i've been using RESTClient https://github.com/chao/RESTClient is the git but just search and add using add on's(also has basic, oauth 1 and 2). Chrome has YARC thats pretty good or if you know SoapUI you can go that way.

File: apis/dispatch.php (hunk 1 above)
Comment (sjpadgett) [reply apis/dispatch.php:46]: Really should not happen and will add throw a 401 or 302 there.

**Pull request merged**